haystack/components/converters/pdfminer.py

```diff
@@ -178,6 +208,15 @@ def run(
 
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
+
+            analysis = self.detect_undecoded_cid_characters(text)
+
+            if analysis["percentage"] > 0:
+                logger.warning(
```

Comment:
We shouldn't use f-strings in logging messages for performance reasons and to help with structured logging. See https://github.com/deepset-ai/haystack/pull/7207 for context.

We need to change this to something like the following
```python
logger.warning(
    "Detected {cid_chars} undecoded CID characters in {total_chars} characters"
    " ({percentage}%) in {source}.", 
    cid_chars=analysis["cid_chars"], total_chars=analysis['total_chars'], percentage=analysis["percentage"], source=source
)
```

Reply:
good point